spdif_rx.c

```diff
@@ -494,7 +533,7 @@ static int _spdif_rx_analyze_capture(spdif_rx_samp_freq_t* samp_freq, bool* inve
         int bit_pos = 0; // position within current dword
         int word_idx = 0;
         while (true) {
-            int r = __builtin_clz((cur) ? ~shift_reg : shift_reg); // leading zeros
+            int r = _spdif_count_leading_zeros((cur) ? ~shift_reg : shift_reg); // leading zeros
```

Comment:
How about simply rescuing for the case of zero input? (not to define `_spdif_count_leading_zeros()`)
```suggestion
            uint32_t pol_shift_reg = cur ? ~shift_reg : shift_reg;
            int r = pol_shift_reg == 0 ? 32 : __builtin_clz(pol_shift_reg); // leading zeros (avoid the case for zero as undefined)
```